### DISCUSSION TITLE
[v3] getUpdateUri() doesn't take into account route parameters

### DISCUSSION THREAD

**Discussion by stancl** [livewire/livewire#6119] [7. Bugs (archived)] [answered]
### Livewire version

v3 beta

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

The installation guide mentions the new `setUpdateRoute()` method as usable for multi-tenancy, however it doesn't seem to take into account route parameters.

In a multi-tenant app, you may have a route like `/{tenant}/livewire/update` when identifying tenants by path:

```php
Livewire::setUpdateRoute(function ($handle) {
    return Route::post('/{tenant}/livewire/update', $handle);
});
```

However this will make `getUpdateUri()` return `/{tenant}/livewire/update`, which in turn gets used on the frontend as a literal string:

https://github.com/livewire/livewire/blob/e3a1a5a70aa037276b98e8d69f19ff194240cb89/src/Mechanisms/FrontendAssets/FrontendAssets.php#L157

The issue is that `getUpdateUri()` uses `$updateRoute->uri` which doesn't handle parameters in any way:

https://github.com/livewire/livewire/blob/32d84c1000d665417a0d9158680161ee9aaacfa1/src/Mechanisms/HandleRequests/HandleRequests.php#L27-L30

A better solution would be to use the `route()` helper, which would allow filling parameters with default values, e.g. with URL generator overrides (this is what we use), or specified default params for routes (which I believe Laravel supports in some way).

Something like:
```php
route($this->updateRoute->getName())
```

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@stancl thanks for reporting! Hmm yeah I think we only tested it with localised routes. We'd be open to PRs to fix this 🙂

**Comment by Arkhas** [comment]
@stancl do you have solution for this ? we have the same issue with our multi-tenancy app :(

**Reply by stancl** [reply]
https://github.com/livewire/livewire/discussions/6119#discussioncomment-8573080

**Comment by stancl** [comment, marked as answer]
This has been fixed in https://github.com/livewire/livewire/pull/6549
